noita_mod/core/files/scripts/ban_perks_events.lua

```diff
@@ -0,0 +1,246 @@
+        local max_count = 0
+
+        local angle = (full_arc) / (col_limit + 1)
+        local inc = angle
+
+        local length = 24
+        local len_inc = 16
+
+        for i, v in ipairs(perks_to_spawn) do
+            max_count = max_count + v[2]
+        end
+
+        for i, v in ipairs(perks_to_spawn) do
+            local pid = v[1]
+            local pcount = v[2]
+
+            if (pcount > 0) then
+                for j = 1, pcount do
+                    -- smooth out the final arc segment angles
+                    if (column == 0 and total_count + col_limit > max_count) then
+                        angle = (full_arc) / (max_count - total_count + 1)
+                        inc = angle
+                    else
+                        total_count = total_count + 1
+                    end
+                    --local degrees = angle * 180 / math.pi
+                    --GamePrint("spawning " .. j .. " at " .. degrees .. "degrees")
+
+                    local px = x + math.cos(angle) * length
+                    local py = y - math.sin(angle) * length
+                    -- Maintain compatibility with NT_sync_perks and NT_team_perks by adding NT_DONT_SHARE
+                    local perk_entity = perk_spawn_with_name(px, py, pid, true)
+                    EntityAddComponent(perk_entity, "VariableStorageComponent",
+                        {
+                            name = "NT_DONT_SHARE",
+                            value_string = "",
+                        })
+
+                    angle = angle + inc
+                    column = column + 1
+
+                    if (column >= col_limit) then
+                        column = 0
+                        col_limit = col_limit + col_size_inc
+                        length = length + len_inc
+
+                        angle = (full_arc) / (col_limit + 1)
+                        inc = angle
+                    end
+                end
+            end
+        end
+    end
+end
+
+local function RemoveBannedPerks(player_id)
+    local perk_was_removed = false
+    -- Not sure how a player might still be nil but play it safe
+    if (player_id ~= nil) then
+        for i, perk_data in ipairs(BannedPerksDataTable()) do
+            local perk_id = perk_data.id
+
+            local flag_name = get_perk_picked_flag_name(perk_id)
+            local pickup_count = tonumber(GlobalsGetValue(flag_name .. "_PICKUP_COUNT", "0"))
+
+            if (GameHasFlagRun(flag_name) or (pickup_count > 0)) and ((perk_data.do_not_remove == nil) or (perk_data.do_not_remove == false)) then
+                perk_was_removed = true
+                print("Removing " .. perk_id)
+                GameRemoveFlagRun(flag_name)
+                pickup_count = 0
+                GlobalsSetValue(flag_name .. "_PICKUP_COUNT", tostring(pickup_count))
+
+                if (perk_data.game_effect ~= nil) then
+                    -- local comp = GameGetGameEffect( player_id, perk_data.game_effect )
```

Comment:
commented code like this should be removed, or add a comment to why it could be beneficial to keep in here